src/prop/minisat/simp/SimpSolver.cc

```diff
@@ -51,23 +51,22 @@ SimpSolver::SimpSolver(Env& env,
                        prop::TheoryProxy* proxy,
                        context::Context* context,
                        context::UserContext* userContext,
-                       prop::PropPfManager* ppm,
                        bool enableIncremental)
-    : Solver(env, proxy, context, userContext, ppm, enableIncremental),
+    : Solver(env, proxy, context, userContext, enableIncremental),
       grow(opt_grow),
       clause_lim(opt_clause_lim),
       subsumption_lim(opt_subsumption_lim),
       simp_garbage_frac(opt_simp_garbage_frac),
       use_asymm(opt_use_asymm),
       // make sure this is not enabled if unsat cores or proofs are on
-      use_rcheck(opt_use_rcheck && !options().smt.produceUnsatCores && !ppm),
+      use_rcheck(opt_use_rcheck && !options().smt.produceUnsatCores),
```

Comment:
See below comment. Although `use_rcheck` is not further used in the constructor, we should also query `env.isSatProofProducing()` to make it uniform.